modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/Sidebar/TasksSidebar/index.tsx

```diff
@@ -64,85 +65,145 @@ const SubtaskCard: React.FC<SubtaskCardProps> = ({expanded, subtask}) => (
 );
 
 const TaskSidebar: React.FC<TaskSidebarProps> = ({expanded}) => {
-	const {displayTask, setSelectedTask, subTasks, tasks} = useSidebarTask();
+	const {
+		filteredTasks,
+		selectedTask,
+		setSelectedTask,
+		subTasks,
+		tasks,
+	} = useSidebarTask();
 
 	const justifyContentSidebar = {
 		'justify-content-between': expanded,
 		'justify-content-center': !expanded,
 	};
+
 	if (!tasks) {
 		return null;
 	}
 
 	return (
 		<div className="task-sidebar">
-			<div
-				className={classNames(
-					'align-items-center d-flex task-sidebar-title',
-					justifyContentSidebar
-				)}
-			>
-				{expanded && (
-					<span className="font-weight-bold">
-						{i18n.translate('tasks')}
-					</span>
-				)}
-
-				<TaskBadge count={tasks.length} />
-			</div>
-
-			<div className="d-flex flex-column mb-1">
-				<div
-					className={classNames('d-flex mb-2', justifyContentSidebar)}
-				>
-					<TaskBadge
-						className={classNames({
-							'mr-3': expanded,
-						})}
-						count={subTasks.length as number}
-					/>
-
-					<div>{displayTask?.task?.name}</div>
-				</div>
-
-				<div>{displayTask?.task?.build?.name}</div>
-
-				<TaskbarProgress
-					displayTotalCompleted
-					items={[
-						[StatusesProgressScore.SELF, 60],
-						[StatusesProgressScore.INCOMPLETE, Number(30)],
-					]}
-					taskbarClassNames={chartClassNames}
-				/>
-			</div>
-
-			{subTasks.map((subtask: TestraySubTask, index) => (
-				<div key={index}>
-					<SubtaskCard
-						expanded={expanded}
-						subtask={{
-							name: subtask?.name,
-							score: subtask.score,
-						}}
-					/>
-				</div>
-			))}
-
-			{tasks.map((taskUser: TestrayTaskUser, index) => (
-				<div className="mb-5 mt-6" key={index}>
-					{expanded && (
+			{!!tasks.length && (
+				<>
+					<div
+						className={classNames(
+							'align-items-center d-flex task-sidebar-title',
+							justifyContentSidebar
+						)}
+					>
+						<span
+							className={classNames('font-weight-bold', {
+								'sidebar-item-hidden': !expanded,
+								'sidebar-item-visible': expanded,
+							})}
```

Comment:
This 

```
{
 								'sidebar-item-hidden': !expanded,
 								'sidebar-item-visible': expanded,
 							}
```

Is used here and in line: 124, you could create a variable and use both places